kazoo/handlers/utils.py

```diff
@@ -194,17 +197,41 @@ def create_tcp_connection(module, address, timeout=None):
     sock = None
 
     while end is None or time.time() < end:
-        try:
-            # if we got a timeout, lets ensure that we decrement the time
-            # otherwise there is no timeout set and we'll call it as such
-            timeout_at = end if end is None else end - time.time()
-            sock = module.create_connection(address, timeout_at)
-            break
-        except Exception as ex:
-            errnum = ex.errno if isinstance(ex, OSError) else ex[0]
-            if errnum == errno.EINTR:
-                continue
-            raise
+        if use_ssl:
+            # Disallow use of SSLv2 and V3 (meaning we require TLSv1.0+)
+            context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+            context.options |= ssl.OP_NO_SSLv2
+            context.options |= ssl.OP_NO_SSLv3
+            context.verify_mode = ssl.CERT_OPTIONAL
+            if ca:
+                context.load_verify_locations(ca)
+            if certfile and keyfile:
+                context.verify_mode = ssl.CERT_REQUIRED
+                context.load_cert_chain(certfile=certfile, keyfile=keyfile)
+            try:
+                # Query the address to get back it's address family
+                addrs = socket.getaddrinfo(address[0], address[1], 0,
+                                           socket.SOCK_STREAM)
+                conn = context.wrap_socket(module.socket(addrs[0][0]))
+                conn.connect(address)
+                sock = conn
+                break
+            except ssl.SSLError:
+                raise
+        else:
+            try:
+                sock = module.create_connection(address, timeout)
```

Comment:
I noticed this wasn't previously present... was it overlooked prior or is there some reason it's needed now?

Reply:
Can you elaborate? Not quite sure what you are referring to. Looks the same as the original author's code, AFAICT.

Reply:
The original code is already copied below in https://github.com/python-zk/kazoo/pull/494/files#diff-6aabced2bdeee5ae3b908caaa504d88eR225

https://github.com/python-zk/kazoo/pull/494/files#diff-6aabced2bdeee5ae3b908caaa504d88eR223 appears to be a copy/paste error.

Reply:
@emonty  / @Ormod can you fix the above?